model/ravelWrap.cc

```diff
@@ -536,7 +536,12 @@ namespace minsky
           }
 #endif
         vector<double> tmp(v.size());
-        for (size_t i=0; i<v.size(); ++i) tmp[i]=v[i];
+        if (v.index().empty()) 
+           for (size_t i=0; i<v.size(); ++i) tmp[i]=v[i];
+        else {
+            size_t inc=0; 
+		    for (auto i: v.index()) tmp[++inc]=v.atHCIndex(i);
```

Comment:
This won't work. The problem is that Ravel::DataCube assumes the data is stored densely, and applies OLAP operations on it assuming that. Here, you're just storing the data sparsely again, in fact, just implementing the same storage structure as on line 540, just less efficiently :)

Reply:
I understand that In terms of ravel evaluation, it does not work. However, copying via a counter over the size of the data, as on line 540, outputs completely differently to plots, for, say, the propertyPriceRealIndex.csv file, in Minsky 2.18.2 than in 2.15.5. I only recover very similar behaviour to version 2.15.5 when I do the above. I know I'm basically taking the sparse data in the Itensor v and packing it densely into tmp, and hence the dataCube. 

My question is then: not only loadDataFromVariable() and loadSliceFromData() need to be rewritten entirely in the full-on tensor implementation, ravelDC_loadData() also does? As a related question, will this require clearing everything in the ravelState and repopulating it in RavelTensor::computeTensor(), along with assigning the cachedResult TensorVal with values that have been appropriatedly transformed from the arg input tensor pointer?